src/widgets/splits/SplitHeader.cpp

```diff
@@ -418,45 +434,46 @@ std::unique_ptr<QMenu> SplitHeader::createMainMenu()
         this->split_->setModerationMode(!this->split_->getModerationMode());
     });
 
-    if (dynamic_cast<TwitchChannel *>(this->split_->getChannel().get()))
+    if (twitchChannel)
     {
         moreMenu->addAction("Show viewer list", this->split_,
                             &Split::showViewerList);
 
         moreMenu->addAction("Subscribe", this->split_, &Split::openSubPage);
 
-        auto action = new QAction(this);
-        action->setText("Notify when live");
-        action->setCheckable(true);
+        // notify live action
+        auto nlAction = new QAction(this);
+        nlAction->setText("Notify when live");
+        nlAction->setCheckable(true);
 
-        QObject::connect(moreMenu, &QMenu::aboutToShow, this, [action, this]() {
-            action->setChecked(getApp()->notifications->isChannelNotified(
-                this->split_->getChannel()->getName(), Platform::Twitch));
-        });
-        action->connect(action, &QAction::triggered, this, [this]() {
+        QObject::connect(
+            moreMenu, &QMenu::aboutToShow, this, [nlAction, this]() {
+                nlAction->setChecked(getApp()->notifications->isChannelNotified(
+                    this->split_->getChannel()->getName(), Platform::Twitch));
+            });
+        nlAction->connect(nlAction, &QAction::triggered, this, [this]() {
             getApp()->notifications->updateChannelNotification(
                 this->split_->getChannel()->getName(), Platform::Twitch);
         });
 
-        moreMenu->addAction(action);
-    }
+        moreMenu->addAction(nlAction);
 
-    if (dynamic_cast<TwitchChannel *>(this->split_->getChannel().get()))
-    {
-        auto action = new QAction(this);
-        action->setText("Mute highlight sound");
-        action->setCheckable(true);
+        // mute highlight action
+        auto mhAction = new QAction(this);
+        mhAction->setText("Mute highlight sound");
+        mhAction->setCheckable(true);
 
-        QObject::connect(moreMenu, &QMenu::aboutToShow, this, [action, this]() {
-            action->setChecked(getSettings()->isMutedChannel(
-                this->split_->getChannel()->getName()));
-        });
-        action->connect(action, &QAction::triggered, this, [this]() {
+        QObject::connect(moreMenu, &QMenu::aboutToShow, this,
+                         [mhAction, this]() {
+                             mhAction->setChecked(getSettings()->isMutedChannel(
+                                 this->split_->getChannel()->getName()));
+                         });
+        mhAction->connect(mhAction, &QAction::triggered, this, [this]() {
             getSettings()->toggleMutedChannel(
                 this->split_->getChannel()->getName());
         });
 
-        moreMenu->addAction(action);
+        moreMenu->addAction(mhAction);
```

Comment:
These changes should definitely be in a separate PR

Reply:
I had a feeling those changes might've been not welcome, though after changing some related code it annoyed me so much I had to change it. You've got a point thoguh and I will revert those changes :+1:

Reply:
Though, I will keep the declaration of `twitchChannel` as it is useful in the PR and makes other lines of code more readable as well.